### PR TITLE
Add Missing Python Bindings and Add/Update Docstrings

### LINKED ISSUE #6427: How to keep drawing geometry independant on view point ?
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### My Question

Hi all,

I've tried to use O3DVisualizer with LineSet but the geometry disapear regarding the view point. Is there a way to force drawing every geometries ? 

Thanks by advance,
Clément

open3d

### PR DESCRIPTION
**Pull request by errissa**
## Type

-   [x ] Bug fix (non-breaking change which fixes an issue): Fixes #6427 and #6413
-   [ ] New feature (non-breaking change which adds functionality). Resolves #
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) Resolves #

## Motivation and Context

Adds missing Python bindings to `Scene` class for controlling object properties such as culling (which provides partial solution to #6427) and to `TreeView` class for retrieving selected widget (addresses #6413)

## Checklist:

-   [ x] I have run `python util/check_style.py --apply` to apply Open3D **code style**
    to my code.
-   [ ] This PR changes Open3D behavior or adds new functionality.
    -   [ ] Both C++ (Doxygen) and Python (Sphinx / Google style) **documentation** is
        updated accordingly.
    -   [ ] I have added or updated C++ and / or Python **unit tests** OR included **test
        results** (e.g. screenshots or numbers) here.
-   [x ] I will follow up and update the code if CI fails.
    
-   [ ] For fork PRs, I have selected **Allow edits from maintainers**.

## Description

- - -
This change is [Reviewable](https://reviewable.io/reviews/isl-org/Open3D/6429)

### CODE REVIEW

File: cpp/pybind/visualization/rendering/rendering.cpp [hunk 1]
```diff
@@ -516,6 +516,24 @@ void pybind_rendering_classes(py::module &m) {
                  "The flags should be ORed from Scene.UPDATE_POINTS_FLAG, "
                  "Scene.UPDATE_NORMALS_FLAG, Scene.UPDATE_COLORS_FLAG, and "
                  "Scene.UPDATE_UV0_FLAG")
+            .def("remove_geometry", &Scene::RemoveGeometry,
```
Comment (ssheorey) [inline cpp/pybind/visualization/rendering/rendering.cpp:519]: Add annotations for arguments for all python bindings. This is necessary for proper documentation from Sphinx, as well as for default values.

e.g.:
```cpp
.def("add_geometry",
                 (bool (Scene::*)(
                         const std::string &, const t::geometry::Geometry &,
                         const MaterialRecord &, const std::string &, size_t)) &
                         Scene::AddGeometry,
                 "name"_a, "geometry"_a, "material"_a,
                 "downsampled_name"_a = "", "downsample_threshold"_a = SIZE_MAX,
                 "Adds a Geometry with a material to the scene")
```

**Review by ssheorey — CHANGES_REQUESTED** [review]

File: cpp/pybind/visualization/rendering/rendering.cpp (hunk 1 above)
Comment (errissa) [reply cpp/pybind/visualization/rendering/rendering.cpp]: Done. Added to all `Scene` pybinds.

File: cpp/pybind/visualization/gui/gui.cpp [hunk 2]
```diff
@@ -1550,6 +1550,11 @@ void pybind_gui_classes(py::module &m) {
             .def("remove_item", &TreeView::RemoveItem,
                  "Removes an item and all its children (if any)")
             .def("clear", &TreeView::Clear, "Removes all items")
+            .def("get_item", &TreeView::GetItem,
```
Comment (ssheorey) [inline cpp/pybind/visualization/gui/gui.cpp:1553]: Document arguments here as well.

File: cpp/pybind/visualization/rendering/rendering.cpp [hunk 3]
```diff
@@ -508,51 +509,83 @@ void pybind_rendering_classes(py::module &m) {
...
+            .def("enable_sun_light", &Scene::EnableSunLight, "enable"_a)
+            .def("set_sun_light", &Scene::SetSunLight, "direction"_a, "color"_a,
+                 "intensity"_a,
+                 "Sets the parameters of the sun light direction, "
                  "color, intensity")
-            .def("add_point_light", &Scene::AddPointLight,
+            .def("add_point_light", &Scene::AddPointLight, "name"_a, "color"_a,
+                 "position"_a, "intensity"_a, "falloff"_a, "cast_shadows"_a,
                  "Adds a point light to the scene: add_point_light(name, "
                  "color, position, intensity, falloff, cast_shadows)")
```
Comment (ssheorey) [inline cpp/pybind/visualization/rendering/rendering.cpp:558]: Can remove the function signature [`add_point_light(name, color, position, intensity, falloff, cast_shadows)` ]
from docstring - it's autogenerated if the arguments are annotated. Also in other places below.
For default values (these should follow default values in the C++ API), the syntax is `"cast_shadows"_a = false`
To document arguments:
e.g. to document `PointCloud.estimate_normals(search_param = KDTreeSearchParamKNN(), fast_normal_computation=True)`
```cpp
docstring::ClassMethodDocInject(
            m, "PointCloud", "estimate_normals",
            {{"search_param",
              "The KDTree search parameters for neighborhood search."},
             {"fast_normal_computation",
              "If true, the normal estimation uses a non-iterative method to "
              "extract the eigenvector from the covariance matrix. This is "
              "faster, but is not as numerical stable."}});
```
These appear nicely in sphinx as:
Screenshot 2023-10-13 at 9 29 57 PM

**Review by ssheorey — COMMENTED** [review]
Thanks Rene. A couple more improvements to the documentation and we can merge.

**Review by ssheorey — APPROVED** [review]

**Pull request merged**